src/evaluators/UnaryExpression.js

```diff
@@ -97,22 +101,33 @@ function evaluateDeleteOperation(expr: Value | Reference, realm: Realm) {
 
 function generateRuntimeCall(ref: Reference | Value, ast: BabelNodeUnaryExpression, strictCode: boolean, realm: Realm) {
   invariant(ref instanceof Reference);
-  let baseValue = Environment.GetBase(realm, ref);
-  invariant(baseValue instanceof Value);
-  let propertyName = Environment.GetReferencedName(realm, ref);
-  invariant(typeof propertyName === "string");
+  let value;
+  let propertyName;
 
-  if (!baseValue.isSimpleObject()) {
-    Leak.leakValue(realm, baseValue, ast.loc);
+  // the base reference might be an EnvironmentRecord or a ObjectValue
+  // if it is a EnvironmentRecord, we don't serialize a member expression below
+  if (Environment.IsPropertyReference(realm, ref)) {
+    value = Environment.GetBase(realm, ref);
+    invariant(value instanceof Value);
+    if (!value.isSimpleObject()) {
```

Comment:
Leak.leakValue doesn't do much in line of optimizing the (probably common) case where value is known not be an object. You could check this with "value.mightBeObject() && !value.isSimpleObject()" or you could add such a check to Leak.leakValue.